### DISCUSSION TITLE
重构demo&doc计划🚀🚀🚀

### DISCUSSION THREAD

**Discussion by Miles-hxy** [jdf2e/nutui-react#2018] [General]
前期为了适应各种版本，共有`doc.md`、`doc.en-US.md`、`doc.zh-TW.md`、`doc.taro.md`四种类型的文档，随着功能的迭代，每次更改demo都需要进行重复的国际化翻译与四份文档的编写，带来了大量的工作负担与一致性隐患。因此在这一方面，本项目已进行mdx项目重构，转变逻辑如下：
![未命名绘图 drawio (1)](https://github.com/jdf2e/nutui-react/assets/99181718/665829ec-af2c-4f83-950e-c1ec794b78ac)

![3f0c6699-0c9b-4b3c-8e7e-56789447a4e7](https://github.com/jdf2e/nutui-react/assets/99181718/e398d4a4-dcf1-4280-864b-7f0c99582a9a)

具体提交PR过程如下：
### fork仓库
访问https://github.com/jdf2e/nutui-react
点击fork仓库到Your repositories(欢迎Star)。
### 克隆代码到本地并运行
单独开辟分支，如`git checkout -b fix/0314/extract-button-demos`
### 具体操作
1. 执行自动拆解与替换脚本
```
node ./scripts/doc-demo-extract-replace.js [组件名]
```
2. 对`demo.tsx`和`demo.taro.tsx`进行组件引入与拆解。去除无关代码。具体效果可参考已完成的组件

- `doc.md`、`doc.en-US.md`、`doc.zh-TW.md`依次引入demos/h5中的全部demo
- `doc.taro.md`引入demos/taro中的全部demo
- 逐个对比抽取出的`demo`与`demo.tsx`对应位置的代码是否一致，若不一致：以`demo.tsx`为标准对抽取出的`demo`进行修改；一致后则替换为引入的`demo`。
- 全部替换完成后，总览`demo`效果，确认无误后删除`interface`以及无关`import`的内容。至此完成对于demo.tsx的替换
- `demo.taro.tsx`的校验步骤与上述一致

<img width="591" alt="demo" src="https://github.com/jdf2e/nutui-react/assets/99181718/c0e0696c-58e3-41e6-a63d-baf16a8e27d5">

3. 脚本执行完成后，对demos文件夹进行格式化，在终端执行
```
npx eslint ./src/packages/[组件名]/demos/ --fix
```
### PR流程
1. commit并push代码，检查自己的代码在pnpm dev与pnpm:taro:h5等环境的demo是否正常，与之前是否一致。
2. 确认无误后，在https://github.com/jdf2e/nutui-react  填写PR说明md并提交Pull request。
3. PR名称规范：`fix([组件名]): demo拆解与规范`  描述勾选如下：
- [x] 站点、文档改进
###  组件认领表
| 组件名称 | 认领人 | 当前进展 | 计划完成时间 |
| --- | --- | --- | --- |
Button|@Alex-huxiyang|已完成https://github.com/jdf2e/nutui-react/pull/1950|2024-3-10
Cell|@Alex-huxiyang|已完成https://github.com/jdf2e/nutui-react/pull/1998|2024-3-10
ConfigProvider|@Alex-huxiyang|已完成https://github.com/jdf2e/nutui-react/pull/2005|2024-3-12
Icon|@Alex-huxiyang|已完成https://github.com/jdf2e/nutui-react/pull/2010|2024-3-13
Image|@Alex-huxiyang|已完成https://github.com/jdf2e/nutui-react/pull/2011|2024-3-13
OverLay|@Alex-huxiyang|已完成https://github.com/jdf2e/nutui-react/pull/2012|2024-3-13
Divider|@Alex-huxiyang|已完成https://github.com/jdf2e/nutui-react/pull/2013|2024-3-13
Grid|@Alex-huxiyang|已完成https://github.com/jdf2e/nutui-react/pull/2014|2024-3-13
Layout|@Alex-huxiyang|已完成https://github.com/jdf2e/nutui-react/pull/2016|2024-3-14
SafeArea|@xiaoyatong|已完成|2024-3-15
Space|@Alex-huxiyang|已完成|2024-3-14
Sticky|@Alex-huxiyang|已完成|2024-3-14
BackTop|@Alex-huxiyang|已完成|2024-3-14
Elevator|@Alex-huxiyang|已完成|2024-3-15
FixedNav| @Alex-huxiyang|已完成|2024-3-18
Navbar|@Alex-huxiyang|已完成|2024-3-18
SideNavBar|@Alex-huxiyang|已完成|2024-3-18
Tabbar|@Alex-huxiyang|已完成|2024-3-18
Tabs|@yeyu98|已完成
Address|@Alex-huxiyang|已完成|2024-3-19
Calendar|@mikasayw|已完成
CalendarCard|@eiinu|已完成|2024-3-14
Cascader|@Alex-huxiyang|已完成
Checkbox|@Alex-huxiyang|已完成
DatePicker|@yeyu98|已完成
Form|@oasis-cloud|已完成
Input|@yeyu98|已完成
InputNumber|@xiaoyatong|已完成https://github.com/jdf2e/nutui-react/pull/2023|2024-3-14
Menu|@Alex-huxiyang|已完成
NumberKeyboard|@ZissyW|已完成
Picker|@Alex-huxiyang|已完成
Radio|@3042503122|已完成
Range|@ShuchenEason|已完成
Rate|@sandra66888|已完成
SearchBar|@joyfully-W|已完成
ShortPassword|@Alex-huxiyang|已完成
Signature|@Alex-huxiyang|已完成
Switch|@jiangjin3323|已完成
TextArea|@Amylee9712|已完成
Uploader|@irisSong|已完成
ActionSheet|@ZissyW|已完成
Badge|@wanglihuaya|已完成
Dialog|@irisSong|已完成
Drag| @eiinu|todo
Empty|@xiaoyatong|已完成
InfiniteLoading|@xiaoyatong|已完成
Loading|@Alex-huxiyang|已完成
NoticeBar|@xiaoyatong|已完成
Notify|@Alex-huxiyang|已完成
Popover|@xiaoyatong|已完成
Popup|@ZissyW|已完成
PullToRefresh|@xiaoyatong|已完成
Skeleton|@wanglihuaya|已完成
Swipe|@oasis-cloud|todo
Toast|@OrdinarySF|已完成
Animate|@eiinu| 已完成
AnimatingNumbers|@Alex-huxiyang|已完成
Audio|@Alex-huxiyang|已完成
Avatar|@Alex-huxiyang|已完成
CircleProgress|@Alex-huxiyang|已完成
Collapse|@eiinu|已完成
Countdown|@Alex-huxiyang|已完成
Ellipsis|@irisSong|已完成
ImagePreview|@Alex-huxiyang|已完成
Indicator|@eiinu| 已完成
List（virtuallist）| @eiinu |已完成
Pagination|@Alex-huxiyang|已完成
Price|@Alex-huxiyang|已完成
Progress|@Alex-huxiyang|已完成
Steps|@eiinu|已完成
Swiper|@Alex-huxiyang|已完成
Table|@Alex-huxiyang|已完成
Tag|@eiinu| 已完成
Tour|@Alex-huxiyang|已完成
Video|@eiinu|已完成
AvatarCropper|@eiinu|已完成
Barrage|@joyfully-W|已完成
Card|@joyfully-W|已完成
TimeSelect|@Alex-huxiyang|已完成
TrendArrow|@sunlanda|已完成
WaterMark|@eiinu| 已完成

**Comment by xiaoyatong** [comment]
InputNumber 已提交pr

**Reply by Miles-hxy** [reply]
已更新

**Comment by eiinu** [comment]
CalenderCard 已提交 PR

**Reply by Miles-hxy** [reply]
收到

**Comment by oasis-cloud** [comment]
Form 组件

**Reply by Miles-hxy** [reply]
done

**Comment by wanglihuaya** [comment]
- Skeleton
- Swipe
- Badge

**Reply by Miles-hxy** [reply]
Done

**Comment by yeyu98** [comment]
- Input
- DatePicker
- Tabs

**Reply by Miles-hxy** [reply]
Done

**Comment by jiangjin3323** [comment]
Switch

**Reply by Miles-hxy** [reply]
Done.

**Comment by OrdinarySF** [comment]
1. loading
2. toast

**Reply by Miles-hxy** [reply]
Done.

**Reply by OrdinarySF** [reply]
toast pr.

**Reply by Miles-hxy** [reply]
Roger.

**Comment by joyfully-W** [comment]
SearchBar  已提交 PR

**Reply by Miles-hxy** [reply]
Done.

**Reply by joyfully-W** [reply]
已重新提交

**Reply by Miles-hxy** [reply]
OK，会定期统一处理PR的

**Comment by sandra66888** [comment]
Rate 组件

**Reply by Miles-hxy** [reply]
Done

**Comment by joyfully-W** [comment]
Card 已提交 PR

**Reply by Miles-hxy** [reply]
done. 记得先认领后做哈，要不挺容易重复的👍

**Comment by sunlanda** [comment]
认领 : trendarrow

**Reply by Miles-hxy** [reply]
Done.

**Comment by joyfully-W** [comment]
Barrage

**Reply by Miles-hxy** [reply]
Done.

**Comment by liuwei1025** [comment]
Cascader
Menu

**Reply by Miles-hxy** [reply]
Done.

**Comment by JoanneXu6677** [comment]
认领 Checkbox

**Reply by Miles-hxy** [reply]
Done

**Comment by 3042503122** [comment]
认领 Radio

**Reply by Miles-hxy** [reply]
Done

**Comment by Amylee9712** [comment]
TextArea

**Reply by Miles-hxy** [reply]
Done.

**Comment by OrdinarySF** [comment]
3. notify

**Reply by Miles-hxy** [reply]
Done.

**Comment by ShuchenEason** [comment]
认领：Range

**Comment by Asdy1011** [comment]
认领：Countdown

**Reply by Miles-hxy** [reply]
Done👍

**Comment by DreamSeeker321** [comment]
认领 Picker

**Reply by xiaoyatong** [reply]
ok

**Comment by Amylee9712** [comment]
textarea已提交PR

**Reply by Miles-hxy** [reply]
好的

**Comment by irisSong** [comment]
ellipsis 已提交pr

**Reply by Miles-hxy** [reply]
好的

**Comment by irisSong** [comment]
认领uploader

**Reply by Miles-hxy** [reply]
好的、

**Comment by mikasayw** [comment]
认领 Calendar

**Reply by Miles-hxy** [reply]
Done.

**Comment by ShuchenEason** [comment]
Range 已提 PR

**Comment by Miles-hxy** [comment]
Got it
